compiler/src/iree/compiler/GlobalOptimization/Passes.cpp

```diff
@@ -68,6 +68,8 @@ void buildGlobalOptimizationPassPipeline(
       .addPass(IREE::Flow::createRaiseSpecialOps)
       .addPass(IREE::Flow::createFoldUnitExtentDimsPass)
       .addPass(IREE::Flow::createFuseDequantizationMatmulPass)
+      // Expand all vectors in vecmat/matvec ops into matrices for tiling.
+      .addPass(createExpandVectorsPass)
```

Comment:
Should we only add this pass when `(transformOptions.options.dataTiling == true`? It seems like the current main purpose of this pass is for data-tiling.

Reply:
Good point. This should be done using `.addPredicatedPass` similarly to SetEncoding, right? (Couldn't find any explicit documentation that that's what it does so just want to make sure)

Reply:
I think so. But let me know if there is a better approach : )

Reply:
Yes, we should only add it when data-tiling is enabled for now. If we want to make it on by default, we should collect some data point.